### ISSUE TITLE
HuggingFaceKerasClassifier Example Notebook + Data-interface Improvements

### ISSUE THREAD

**Issue by jwmueller** [cleanlab/cleanlab#372]
First address: https://github.com/cleanlab/cleanlab/issues/433
Before starting work on this issue!

Get familiar with this PR: https://github.com/cleanlab/cleanlab/pull/247

**Goal**:  Transform the code from comment description in that above PR into a .ipynb notebook that will be added to: [github.com/cleanlab/examples](http://github.com/cleanlab/examples)

The code is pretty unwieldy because at that time, cleanlab’s CleanLearning only supported numpy arrays.  We have since added a PR to make CleanLearning compatible with tensorflow/torch Datasets and other non-numpy objects:  https://github.com/cleanlab/cleanlab/pull/311

TODOs:
- [x] Try to update this code: https://github.com/cleanlab/cleanlab/blob/master/cleanlab/experimental/huggingface_keras_classifier.py
to interface with dataset in a nicer way (rather than having dataset only consist of indices as it is currently). May choose to remove this file entirely if our KerasWrapper can support the same thing.
- [x] Make modifications to CleanLearning's torch/tensorflow dataset support as necessary to get ^ working well. 
Eg. for tensorflow datasets, we may want to add more checks if dataset is being shuffled or has other weird properties.
- [x] Create notebook demonstrating the usage of the improved HuggingFaceKerasClassifier (or KerasWrapper) and make PR to examples/ repo. 
- [x] Make documentation for this class appear in docs.cleanlab.ai (if we keep it, otherwise make KerasWrapper visible in the docs).

For reference here's the original example code your notebook can be based off of:
```python
import tensorflow as tf
import tensorflow_datasets as tfds
import numpy as np
from cleanlab.experimental.huggingface_keras_classifier import HuggingfaceKerasClassifier
from cleanlab.classification import CleanLearning
from transformers import AutoTokenizer, TFAutoModel
import os
from sklearn.metrics import accuracy_score


os.environ['TOKENIZERS_PARALLELISM']='false'

def build_model(model_name:str, max_len:int, n_classes:int):
    # 1 - Input ids and attention mask as input to NN
    input_ids = tf.keras.layers.Input(
        shape=(max_len,), dtype='int32', name='input_ids')
    attention_mask = tf.keras.layers.Input(
        shape=(max_len,), dtype='int32', name='attention_mask')

    # 2 - Get bert Main layer and add it to the NN, passing input ids and attention mask as inputs
    bert_layer = TFAutoModel.from_pretrained(model_name)
    
    layer = bert_layer(input_ids=input_ids, attention_mask=attention_mask)[1]
    # embeddings = tf.keras.layers.Embedding(30_522 + 1, 128)(input_ids)
    # flatten = tf.keras.layers.Flatten()(embeddings)
    output_layer = tf.keras.layers.Dense(
      n_classes, activation='sigmoid')(layer)

    # model instance
    model = tf.keras.Model(inputs=[input_ids, attention_mask], outputs=output_layer)
    model.summary()
    return model

MODEL_NAME = "bert-base-uncased"
train = tfds.load('imdb_reviews', split='train', shuffle_files=True)
val = tfds.load('imdb_reviews', split='test', shuffle_files=True)
train_df = tfds.as_dataframe(train)[0:1000]
train_df['text'] = train_df['text'].apply(lambda x: x.decode('utf-8'))
val_df = tfds.as_dataframe(val)[0:1000]
val_df['text'] = val_df['text'].apply(lambda x: x.decode('utf-8'))
tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME)

train_input = tokenizer(
  train_df['text'].to_list(),
  padding='max_length',
  truncation=True,
  max_length=20,
  return_tensors='tf')

val_input = tokenizer(
  val_df['text'].to_list(),
  padding='max_length',
  truncation=True,
  max_length=20,
  return_tensors='tf',
  return_token_type_ids=False)

early_stopping = tf.keras.callbacks.EarlyStopping(
        monitor='val_accuracy', mode='max', verbose=1, patience=3, restore_best_weights=True)

val_dataset = tf.data.Dataset.from_tensor_slices(
                (
                    dict(val_input),
                    val_df['label'],
                )
            ).batch(32)

model = HuggingfaceKerasClassifier(
    # --- model function parameters ---
    model=build_model,
    model_name=MODEL_NAME,
    max_len=20,
    n_classes=1,
    # --- HuggingfaceKerasClassifier Parameters ---
    seq_len=20,
    train_input=dict(train_input),
    #   --- TF training Parameters ---
    optimizer=tf.keras.optimizers.Adam(2e-5),
    loss=tf.keras.losses.BinaryCrossentropy(),
    metrics=['accuracy'],
    epochs=10,
    batch_size=64,
    shuffle=True,
    callbacks=[early_stopping],
    verbose=True
)

print('Model Created.')
lnl = CleanLearning(clf=model, cv_n_folds=3)

print('Training CleanLearning model...')
training_ids = np.array(list(range(len(train_input['input_ids']))))

# Note that the validation dataset provided in the clf_kwargs is not related 
# to the validation set used internally by cleanlab. In this case, the validation 
# set is required by the early stopping callback used during the trainings.
lnl.fit(training_ids, np.array(train_df['label']), clf_kwargs={'validation_data': val_dataset})
print('Model trained.')

predictions = lnl.predict(test_input = val_input)
print('Accuracy on val data: ', accuracy_score(val_df['label'], predictions))
```
